universal/src/logging/json_string.cpp

```diff
@@ -0,0 +1,50 @@
+#include "userver/logging/json_string.hpp"
+
+#include <userver/logging/log.hpp>
+#include <userver/utils/assert.hpp>
+
+#include <algorithm>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace logging {
+
+namespace {
+constexpr std::string_view kNull = "null";
+
+bool DebugValid(std::string_view json) {
+  try {
+    formats::json::FromString(json);
+  } catch (const formats::json::ParseException& error) {
+    LOG_ERROR() << "Invalid json: " << json << ", error: " << error;
+    return false;
+  }
+  return true;
+}
+
+}  // namespace
+
+JsonString::JsonString(const formats::json::Value& value)
+    : json_{ToString(value)} {
+  // ToString builds one line string by RapidJson
+}
+
+JsonString::JsonString(std::string json) noexcept : json_{std::move(json)} {
+  UASSERT(DebugValid(json_));
+
+  // Remove extra new lines from user provided json string
```

Comment:
Add a note that this is to avoid additional escaping in TSKV format

Reply:
fixed